src/test/java/io/rtr/conduit/integration/ShutdownHandlerIntegrationTest.java

```diff
@@ -0,0 +1,118 @@
+package io.rtr.conduit.integration;
+
+import io.rtr.conduit.amqp.AMQPMessageBundle;
+import io.rtr.conduit.amqp.consumer.Consumer;
+import io.rtr.conduit.amqp.publisher.Publisher;
+import io.rtr.conduit.util.RecordingAmqpCallbackHandler;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.RabbitMQContainer;
+import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+@Testcontainers
+public class ShutdownHandlerIntegrationTest {
```

Comment:
This test was extracted in essence from AMQPIntegrationIT.

The test aims to verify the behaviour of Conduit and the RabbitMQ client when an temporary outage happens. We are using https://github.com/Shopify/toxiproxy and testcontainer's customisation for it https://www.testcontainers.org/modules/toxiproxy/ to simulate a network outage.

Reply:
Not the end of this kind of testing, this harness should give us a platform to test temporary outages or what is happening with dynamic/non-dynamic queues disappearing, etc.